code/datums/antagonists/datum_wizard.dm

```diff
@@ -0,0 +1,175 @@
+	var/special_role = "wizard"
+	var/list/objectives_given
+
+/datum/antagonist/wizard/apprentice
+	name = "Wizard Apprentice"
+	special_role = "apprentice"
+	var/school = "robeless"
+	var/mob/living/carbon/human/summoner
+
+/datum/antagonist/wizard/can_be_owned()
+	if(!ishuman(owner.current))
+		return FALSE
+	return ..()
+
+/datum/antagonist/wizard/on_gain()
+	SSticker.mode.wizards += owner
+	owner.special_role = special_role
+	return ..()
+
+/datum/antagonist/wizard/proc/forge_wizard_objectives()
+	switch(rand(1,100))
+		if(1 to 30)
+			var/datum/objective/assassinate/kill_objective = new
+			kill_objective.owner = owner
+			kill_objective.find_target()
+			add_objective(kill_objective)
+			
+			forge_escape_objective()
+
+		if(31 to 60)
+			var/datum/objective/steal/steal_objective = new
+			steal_objective.owner = owner
+			steal_objective.find_target()
+			add_objective(steal_objective)
+			
+			forge_escape_objective()
+
+		if(61 to 85)
+			var/datum/objective/assassinate/kill_objective = new
+			kill_objective.owner = owner
+			kill_objective.find_target()
+			add_objective(kill_objective)
+
+			var/datum/objective/steal/steal_objective = new
+			steal_objective.owner = owner
+			steal_objective.find_target()
+			add_objective(steal_objective)
+			
+			forge_escape_objective()
+
+		else if(!(locate(/datum/objective/hijack) in owner.objectives))
+			var/datum/objective/hijack/hijack_objective = new
+			hijack_objective.owner = owner
+			add_objective(hijack_objective)
+	return
+
+/datum/antagonist/wizard/proc/forge_escape_objective()
+	if(!(locate(/datum/objective/escape) in owner.objectives))
+		var/datum/objective/escape/escape_objective = new
+		escape_objective.owner = owner
+		add_objective(escape_objective)
+
+/datum/antagonist/wizard/apply_innate_effects()
+	name_wizard()
+	forge_wizard_objectives()
+	var/mob/living/carbon/human/H = owner.current
+	H.equipOutfit(/datum/outfit/wizard)
+	finalize_wizard()
+	return
+
+/datum/antagonist/wizard/proc/finalize_wizard()
+	to_chat(owner, "You will find a list of available spells in your spell book. Choose your magic arsenal carefully.")
+	to_chat(owner, "The spellbook is bound to you, and others cannot use it.")
+	to_chat(owner, "In your pockets you will find a teleport scroll. Use it as needed.")
+	owner.store_memory("<B>Remember:</B> do not forget to prepare your spells.")
+	SSticker.mode.update_wiz_icons_added(owner)
+	return
+
+/datum/antagonist/wizard/greet()
+	to_chat(owner, "<span class='boldannounce'>You are the Space Wizard!</span>")
+	to_chat(owner, "<B>The Space Wizards Federation has given you the following tasks:</B>")
+	owner.announce_objectives()
+	return
+
+/datum/antagonist/wizard/proc/name_wizard()
+	var/randomname = "[pick(GLOB.wizard_first)] [pick(GLOB.wizard_second)]"
+	var/newname = copytext(sanitize(input(owner.current, "You are the Space Wizard. Would you like to change your name to something else?", "Name change", randomname) as null|text),1,MAX_NAME_LEN)
```

Comment:
Is it safe to have a sleeping proc in antag datum setup? I think you should at least split the naming stuff off into an asyncronous proc